eyecite/utils.py

```diff
@@ -76,7 +78,8 @@ class HashableDict(dict):
     """Dict that works as an attribute of a hashable dataclass."""
 
     def __hash__(self):
-        return hash(frozenset(self.items()))
+        # dict -> sorted dict -> json (str) -> sha256 (int)
+        return str_sha256(json.dumps(self, sort_keys=True).encode("utf8"))
```

Comment:
Yes I did it because python dictionaries are not ordered by default, which means that two dicts with the exact same key/values could lead to two different json strings if their keys are not ordered.